static/js/containers/UserPage_test.js

```diff
@@ -145,17 +155,52 @@ describe("UserPage", () => {
           let addButton = div.getElementsByClassName('profile-tab-card')[1].
             getElementsByClassName('profile-add-button')[0];
 
-          return listenForActions([
+          let updatedProfile = _.cloneDeep(USER_PROFILE_RESPONSE);
+          let entry = generateNewEducation();
+          entry.graduation_date = "1999-12-01";
+          entry.school_name = "A School";
+          entry.school_country = "AF";
+          entry.school_state_or_territory = "AF-BAL";
+          entry.school_city = "FoobarVille";
+          updatedProfile.work_history.push(entry);
+
+          patchUserProfileStub.throws("Invalid arguments");
+          patchUserProfileStub.withArgs(SETTINGS.username, updatedProfile).returns(
+            Promise.resolve(updatedProfile)
+          );
+
+          let expectedActions = [
             START_PROFILE_EDIT,
-            UPDATE_PROFILE,
             SET_EDUCATION_DIALOG_INDEX,
             SET_EDUCATION_DIALOG_VISIBILITY,
             SET_EDUCATION_DEGREE_LEVEL,
-          ], () => {
+            UPDATE_PROFILE_VALIDATION,
+            REQUEST_PATCH_USER_PROFILE,
+          ];
+          for (let i = 0; i < 10; i++) {
+            expectedActions = expectedActions.concat(UPDATE_PROFILE);
```

Comment:
In a for loop we should avoid using `concat` this way since it will make a copy for each loop. Can you change this and the loop in the next test to use `push` instead?